core/src/main/scala/kafka/server/metadata/KRaftMetadataCache.scala

```diff
@@ -550,8 +550,10 @@ class KRaftMetadataCache(
   override def features(): FinalizedFeatures = {
     val image = _currentImage
     val finalizedFeatures = new java.util.HashMap[String, java.lang.Short](image.features().finalizedVersions())
-    finalizedFeatures.put(KRaftVersion.FEATURE_NAME, kraftVersionSupplier.get().featureLevel())
-
+    val kraftVersionLevel = kraftVersionSupplier.get().featureLevel()
+    if (kraftVersionLevel > 0) {
```

Comment:
Is this a kind of bug fix?

Reply:
Features are defaulted to level 0, so we don't need that entry in the map. Since `kraftVersion` can now have feature level 1 to support KIP-853, we should add that entry to the map.